src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -340,7 +340,7 @@ private IntPtr marshalHeaders(IEnumerable<Header> headers)
             {
                 if (valOffset != 0 || valLength != 0)
                 {
-                    throw new ArgumentException("valOffset and valLength parameters must be 0 when producing null values.");
+                    throw new InvalidOperationException("valOffset and valLength parameters must be 0 when producing null values.");
```

Comment:
this sounds like an argumentException to me

Reply:
i was on the fence. will revert.